python/dgl/graphbolt/impl/sampled_subgraph_impl.py

```diff
@@ -53,22 +55,34 @@ def __post_init__(self):
                     isinstance(etype, str)
                     and len(etype_str_to_tuple(etype)) == 3
                 ), "Edge type should be a string in format of str:str:str."
-                assert (
-                    pair.indptr is not None and pair.indices is not None
-                ), "Node pair should be have indptr and indice."
-                assert isinstance(pair.indptr, torch.Tensor) and isinstance(
-                    pair.indices, torch.Tensor
-                ), "Nodes in pairs should be of type torch.Tensor."
+                assert pair.indptr is not None and isinstance(
+                    pair.indptr, torch.Tensor
+                ), "Node pair should be have indptr of type torch.Tensor."
+                # For CUDA, indices may be None because it will be fetched later.
```

Comment:
I am wondering how much performance we can gain by asynchronously fetch indices, esp, we fetch original_edge_ids in the sync way.

Please make balance on performance gain and code structure complexity.

Reply:
25%

Reply:
25% is a lot so we needed to do it.